app/src/main/java/org/simple/clinic/home/patients/PatientsTabUpdate.kt

```diff
@@ -100,4 +105,42 @@ class PatientsTabUpdate(private val isNotifyAppUpdateAvailableV2Enabled: Boolean
     else
       noChange()
   }
+
+  private fun showAppUpdateAvailableMessageBasedOnFeatureFlag(
+      model: PatientsTabModel,
+      event: RequiredInfoForShowingAppUpdateLoaded
+  ): Next<PatientsTabModel, PatientsTabEffect> {
+    return if (isNotifyAppUpdateAvailableV2Enabled) {
+      appUpdateNudgeBasedOnPriority(model, event)
+    } else {
+      showAppUpdateAvailableMessage(event)
+    }
+  }
+
+  private fun appUpdateNudgeBasedOnPriority(
+      model: PatientsTabModel,
+      event: RequiredInfoForShowingAppUpdateLoaded
+  ): Next<PatientsTabModel, PatientsTabEffect> {
+    return when (event.appUpdateNudgePriority!!) {
+      LIGHT, MEDIUM -> showAppUpdateAvailableDialog(model, event)
+      CRITICAL, CRITICAL_SECURITY -> noChange()
+    }
+  }
+
+  private fun showAppUpdateAvailableDialog(
+      model: PatientsTabModel,
+      event: RequiredInfoForShowingAppUpdateLoaded
+  ): Next<PatientsTabModel, PatientsTabEffect> {
+    val appUpdateLastShownOn = event.appUpdateLastShownOn
+    val currentDate = event.currentDate
+
+    val hasADayPassedSinceUpdateLastShown = appUpdateLastShownOn.isBefore(currentDate)
+
+    val shouldShowAppUpdate = event.isAppUpdateAvailable && hasADayPassedSinceUpdateLastShown && model.hasAppUpdateNudgePriority
+
+    return if (shouldShowAppUpdate)
+      next(model.appUpdateNudgePriorityUpdated(event.appUpdateNudgePriority!!), ShowAppUpdateAvailable, TouchAppUpdateShownAtTime)
```

Comment:
This is still the older app update dialog but we are updating the nudge priority right?

Reply:
yep!

Reply:
As we were going to remove the old one, I didn't think of refactoring this much.